### PR TITLE
fix: [CDS-76698]: remove field from yaml which has empty value in jira/service now steps

### PR DESCRIPTION
**Pull request by satyam-harness**
Ask -> In jira create/update and serviceNow create/update steps, fields are populated depending upon projectId, issueType/issueKey etc. These field are not mandatory, so they can be saved without giving any value which will appear as empty string in yaml. Ask was to remove such fields which have empty value.

Approach -> In the `processFormData` method of above steps, put condition to check on the value of fields, if they are empty string, dont push them to yaml

#### Screenshots

https://github.com/harness/harness-core-ui/assets/96036463/aaf74f12-915f-4588-a979-85be707207b1

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraUpdate/helper.ts [hunk 1]
```diff
@@ -22,11 +22,21 @@ export const processFieldsForSubmit = (values: JiraUpdateData): JiraCreateFieldT
...
     const alreadyPresent = toReturn.find(field => field.name === kvField.name)
-    if (!alreadyPresent) {
+    const value =
+      typeof kvField.value === 'string' || typeof kvField.value === 'number'
+        ? kvField.value
+        : Array.isArray(kvField.value)
+        ? (kvField.value as MultiSelectOption[]).map(opt => opt.value.toString()).join(',')
+        : typeof kvField.value === 'object'
+        ? (kvField.value as SelectOption).value?.toString()
+        : ''
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JiraUpdate/helper.ts:38]: Can we move to a helper util which can be used both for JiraUpdate and ServiceNowCreate below.

Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/JiraUpdate/helper.ts]: addressed

**Comment by satyam-harness** [comment]
retrigger cypress-cd

**Comment by satyam-harness** [comment]
retrigger cypress-cv

**Comment by satyam-harness** [comment]
retrigger cypress-pipeline

**Comment by satyam-harness** [comment]
retrigger cypress-pipeline

**Comment by satyam-harness** [comment]
retrigger cypress-cv

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/StepsHelper.ts [hunk 2]
```diff
@@ -61,3 +66,13 @@ export function isRetryIntervalGreaterThanTimeout(formikValue: JiraApprovalData
   }
   return false
 }
+
+export const getkvFieldValue = (kvField: FieldType): string | number => {
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/StepsHelper.ts:70]: nit: getKVFieldValue

**Review by PratyushGar — APPROVED** [review]

**Comment by satyam-harness** [comment]
retrigger cypress-pipeline

**Pull request merged**